autogen.sh

```diff
@@ -2,6 +2,14 @@
 
 test -d ./config || mkdir ./config
 
+if [ ! -f test/Makefile.am ] || [ ! -f test/apps/Makefile.am ] ||
+   [ ! -f test/include/Makefile.am ] || [ ! -f test/performance/Makefile.am ] ||
+   [ ! -f test/shmemx/Makefile.am ]  || [ ! -f test/spec-example/Makefile.am ] ||
+   [ ! -f test/unit/Makefile.am ];
+then
+  test -f modules/tests-sos/autogen.sh || { echo "ERROR: The test directory is empty. Please run the following command to download the SOS tests:"; echo "git submodule update --init"; exit 1; }
```

Comment:
A naive question.. is there a way to auto download it?

Reply:
We could definitely auto download it here in `autogen.sh` but based on a few other projects (e.g. ompi, mpich, oshmpi, raspberry pi, etc) it seems this is not standard practice.  Possibly because users (developers, primarily) may not always want to `update --init` submodules when running `autogen.sh`.  On the other hand, if no files are there, maybe we should do it automatically, but seems like it's probably best to let a human figure it out.  Thoughts?

Reply:
I see the reason.. a user may not want to auto download in case there is a local version there already downloaded and possibly updated with newer changes. But, as you also mentioned, for the latter case (no files are there), we can do it auto, right? Is there any drawbacks or cases where it may not help?